clients/src/main/java/org/apache/kafka/clients/consumer/internals/PartitionAssignor.java

```diff
@@ -80,15 +80,15 @@ default void onAssignment(Assignment assignment, int generation) {
     }
 
     /**
-     * Indicate which rebalance protocol this assignor can would work with;
+     * Indicate which rebalance protocol this assignor works with;
      * By default it should always work with {@link RebalanceProtocol#EAGER}.
      */
     default List<RebalanceProtocol> supportedProtocols() {
         return Collections.singletonList(RebalanceProtocol.EAGER);
     }
 
     /**
-     * Return the version of the assignor which indicate how the user metadata encodings
+     * Return the version of the assignor which indicates how the user metadata encodings
```

Comment:
I think I mentioned this before as well, but can we elaborate on the use of this version? This is very vague.

Reply:
The version of the assignor would be used at the broker-side to select the leader as the one with the highest version, expecting it be able to decode assignor's older subscription user metadata.

I will update the javadoc as well after the broker's logic has been updated accordingly.